src/_P039_Thermosensors.ino

```diff
@@ -772,29 +802,18 @@ float readMax6675(struct EventStruct *event)
 float readMax31855(struct EventStruct *event)
 {
   P039_data_struct *P039_data = static_cast<P039_data_struct *>(getPluginTaskData(event->TaskIndex));
-  
-  union {
-      uint8_t messageBuffer[4] = {0x00, 0x00, 0x00, 0x00};
-      uint32_t value;
-  } u_mB;
-  
-  uint8_t CS_pin_no = get_SPI_CS_Pin(event);
 
-  // u_mB.messageBuffer[4] = {0x00, 0x00, 0x00, 0x00};
-
-  // "transfer" 0x0 and read the 32 Bit conversion register from the Chip
+  uint8_t messageBuffer[4] = {0x00, 0x00, 0x00, 0x00};
 
-  transfer_n_ByteSPI(CS_pin_no, 4, &u_mB.messageBuffer[0]);
-  uint16_t rawvalue = u_mB.value;
-
-  // uint32_t rawvalue = read_n_ByteitRegister(CS_pin_no, 0x0000);
+  uint8_t CS_pin_no = get_SPI_CS_Pin(event);
 
-  // // Shift MSB 16 Bits to the left
-  // rawvalue <<= 16;
+  // "transfer" 0x0 and read the 32 Bit conversion register from the Chip
+  transfer_n_ByteSPI(CS_pin_no, 4, &messageBuffer[0]);
 
-  // // "transfer" 0x0 and read the LSB Data from the Chip
-  // rawvalue |= read16BitRegister(CS_pin_no, 0x0000);
+  // merge rawvalue from 4 bytes of messageBuffer
+  uint32_t rawvalue = ((messageBuffer[0] << 24) | (messageBuffer[1] << 16) | (messageBuffer[2] << 8) | messageBuffer[3]);
```

Comment:
Not 100% sure if this bitshifting will work.
The type of the `messageBuffer` is 8 bits unsigned int.
I am not 100% certain what the compiler expects the type to be for `operator<<`.
If it is cast to uint32_t, before shifting, then it will work just fine.

```c++
uint32_t rawvalue = ((static_cast<uint32_t>(messageBuffer[0]) << 24) | (static_cast<uint32_t>(messageBuffer[1]) << 16) | (static_cast<uint32_t>(messageBuffer[2]) << 8) | static_cast<uint32_t>(messageBuffer[3]));
```
This is more explicit, albeit rather hard to read.

Reply:
Good finding - in regards of compiler, sometimes i program like "in god i trust" - i will update this one for my nest push

Reply:
I usually also don't mind a lot on those, but the problem is that if it gives unexpected results (e.g. when it is not cast to uint32_t but to int32_t) those bugs are hard to spot later.
I know for a fact bitshifts like this will not work on > 32 bit values. (as in, took me quite a lot of frustration to find it back then)

Reply:
Solved in upcoming push !